### ISSUE TITLE
[Feature]: support controllable ttl for caching on proxy 

### ISSUE THREAD

**Issue by krrishdholakia** [BerriAI/litellm#2649]
### The Feature

allow user to control the ttl for caching on the proxy 

### Motivation, pitch

user request

**Comment by hoangleitvn** [comment]
I also faced caching on proxy issues when dealing with the user budget. Right now we don't have any way to configure cache TTL at all. When I use `user/update` API to adjust the budget and spend amount, the proxy is still using the old value and it makes the completion call is still using the old value. I am not using Redis at all, for now, I have to restart the proxy to reset the cache.

**Comment by ghost** [comment]
hey @hoang-innomize the user is cached for 60s, to reduce the amount of db calls made in prod. What would you want here?

**Comment by hoangleitvn** [comment]
@krrishdholakia It seems the cache TTL is 10 minutes not 60s as you have mentioned above

**Comment by ghost** [comment]
You're right. That's my bad - we should have these be an enum viewable at the top-level so it's easier to find this information. 

Will work on a refactor here.

**Comment by ghost** [comment]
fix pushed @hoang-innomize https://github.com/BerriAI/litellm/commit/26c0ed0f2d1fa22d80a84ec2a9d4a21e6cdd390e

**Comment by hoangleitvn** [comment]
Thanks. We could use 10m as the cache TTL for user information, I believe it will also for the budget info as well. If we also can update the logic for the update user API to reset the cache, that will be helpful. That way, when we call API to adjust budget, we get the cache cleared and can get the latest budget instead of waiting for 10mins

**Comment by ghost** [comment]
Hey @hoang-innomize how would you want to control this?

**Comment by hoangleitvn** [comment]
@krrishdholakia I believe we should update the update user function to clear the user information cache

async def user_update(data: UpdateUserRequest):

**Comment by ghost** [comment]
Got it
